docs/components/tabset/tabset-demo.js

```diff
@@ -0,0 +1,106 @@
+
+if (document.getElementById('vue-dynamicTabsetDemo')) {
+    new Vue({
+        el: '#vue-dynamicTabsetDemo',
+        data: getDefaultData(),
+        updated: function () {
+            // keep correct tab/panel pair open when a "tab" is added
+            this.$refs.tabset.update();
+
+            // keep currentTab in bounds
+            if (this.currentTab >= this.tabs.length) {
+                this.currentTab = (this.tabs.length ? this.tabs.length - 1 : 0);
+            }
+        },
+        methods: {
+            removeTab: function (dir) {
+                switch (dir) {
+                    case 'start':
+                        this.tabs.shift();
+                        break;
+                    case 'end':
+                        this.tabs.pop();
+                        break;
+                    default:
+                        // do nothing
+                        break;
+                }
+            },
+            addTab: function (dir) {
+                let id = this.nextID++;
+
+                switch (dir) {
+                    case 'start':
+                        this.tabs.unshift({ id });
+                        break;
+                    case 'end':
+                        this.tabs.push({ id });
+                        break;
+                    default:
+                        // do nothing
+                        break;
+                }
+            },
+            reset: function () {
+                Object.assign(this.$data, getDefaultData());
+            },
+        },
+        computed: {
+            _tabs: function () {
+                return this.tabs.map((tab, idx) => {
+                    let html = `<hx-tab id="tab-${tab.id}"`;
+                    if (idx === this.currentTab) {
+                        html += ' current="true"';
+                    }
+                    html += '></hx-tab>';
+                    return html;
+                });
+            },
+            _tabpanels: function () {
+                return this.tabs.map((tab, idx) => {
+                    let html = `<hx-tabpanel id="panel-${tab.id}"`;
+                    if (idx === this.currentTab) {
+                        html += ' open';
+                    }
+                    html += '></hx-tabpanel>';
+                    return html;
+                });
+            },
+            _tablist: function () {
+                return this._tabs.reduce((all, tab) => {
+                    return `${all}\n    ${tab}`;
+                }, '');
+            },
+            _tabcontent: function () {
+                return this._tabpanels.reduce((all, panel) => {
+                    return `${all}\n    ${panel}`;
+                }, '');
+            },
+            snippet: function () {
+                return Util.snippet(`
+<hx-tabset current-tab="${this.currentTab}">
+  <hx-tablist>
+    ${this._tablist}
+  </hx-tablist>
+  <hx-tabcontent>
+    ${this._tabcontent}
+  </hx-tabcontent>
+</hx-tabset>
```

Comment:
Indentation is intentional, because `Util.snippet()` isn't smart enough to re-indent HTML elements.